evalml/data_checks/no_variance_data_check.py

```diff
@@ -50,32 +51,34 @@ def validate(self, X, y):
             y (pd.Series): The target data.
 
         Returns:
-            list (DataCheckWarning or DataCheckError): List of warnings/errors corresponding to features or target with no variance.
+            dict (DataCheckWarning or DataCheckError): dict of warnings/errors corresponding to features or target with no variance.
         """
+        messages = {
+            "warnings": [],
+            "errors": []
+        }
+
         if not isinstance(X, pd.DataFrame):
             X = pd.DataFrame(X)
-
         if not isinstance(y, pd.Series):
             y = pd.Series(y)
 
         unique_counts = X.nunique(dropna=self._dropnan).to_dict()
         any_nulls = (X.isnull().any()).to_dict()
-
-        messages = []
-
         for name in unique_counts:
             message = self._check_for_errors(name, unique_counts[name], any_nulls[name])
-
             if message:
-                messages.append(message)
-
+                if message.message_type == DataCheckMessageType.ERROR:
+                    messages["errors"].append(message.to_dict())
+                elif message.message_type == DataCheckMessageType.WARNING:
+                    messages["warnings"].append(message.to_dict())
```

Comment:
What about
```python
for name in unique_counts:
    message = self._check_for_errors(name, unique_counts[name], any_nulls[name])
    if not message:
        continue
    messages[message.message_type].append(message.to_dict())
```
---
And, is there any way we can make this easier to implement? What if we defined this helper in `DataCheck`
```python
@staticmethod
def add_message(message, messages):
    if message.message_type not in messages:
        messages[message.message_type] = []
    messages[message.message_type].append(message.to_dict())
```
and then here in the no-variance data check:
```python
for name in unique_counts:
    message = self._check_for_errors(name, unique_counts[name], any_nulls[name])
    DataCheck.add_message(message, messages)
```